cypress/integration/console/integrations/webhooks/create-with-template.spec.js

```diff
@@ -0,0 +1,104 @@
+    ids: {
+      application_id: appId,
+    },
+  }
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+    cy.createApplication(application, userId)
+  })
+
+  beforeEach(() => {
+    cy.loginConsole({ user_id: userId, password: user.password })
+    cy.fixture('console/application/integrations/webhook/template.json').then(webhookTemplate => {
+      cy.intercept(
+        'GET',
+        `/api/v3/as/webhook-templates?field_mask=base_url,create_downlink_api_key,description,documentation_url,downlink_ack,downlink_failed,downlink_nack,downlink_queue_invalidated,downlink_queued,downlink_sent,fields,format,headers,ids,info_url,join_accept,location_solved,logo_url,name,service_data,uplink_message`,
+        webhookTemplate,
+      )
+      cy.visit(
+        `${Cypress.config(
+          'consoleRootPath',
+        )}/applications/${appId}/integrations/webhooks/add/template`,
+      )
+      cy.findByText('Akenza Core').click()
+    })
+  })
+
+  it('displays UI elements in place', () => {
+    cy.findByRole('heading', { name: 'Add custom webhook' }).should('be.visible')
+    cy.findByRole('heading', { name: 'Template information' }).should('be.visible')
+    cy.findByRole('heading', { name: 'Template settings' }).should('be.visible')
+    cy.findByLabelText('Webhook ID')
+      .should('have.attr', 'placeholder')
+      .and('eq', 'my-new-akenza-webhook')
+    cy.findByLabelText('Domain Secret').should('exist')
+    cy.findByText('Akenza Core domain secret')
+    cy.findByLabelText('Device ID').should('exist')
+    cy.findByText('Akenza Core device ID')
+    cy.findByRole('button', { name: 'Create akenza core webhook' }).should('be.visible')
+  })
+
+  it('validates before submitting an empty form', () => {
+    cy.findByRole('button', { name: 'Create akenza core webhook' }).click()
+
+    cy.findErrorByLabelText('Webhook ID')
+      .should('contain.text', 'Webhook ID is required')
+      .and('be.visible')
+    cy.findErrorByLabelText('Domain Secret')
+      .should('contain.text', 'Domain Secret is required')
+      .and('be.visible')
+    cy.findErrorByLabelText('Device ID')
+      .should('contain.text', 'Device ID is required')
+      .and('be.visible')
+    cy.location('pathname').should(
+      'eq',
+      `${Cypress.config(
+        'consoleRootPath',
+      )}/applications/${appId}/integrations/webhooks/add/template/akenza`,
+    )
+  })
+
+  it('succeeds adding webhook', () => {
+    const webhook = {
+      id: 'my-new-akenza-webhook',
+      domainSecret: 'secret',
+      deviceId: 'end-device-id',
+    }
+    cy.findByLabelText('Webhook ID').type(webhook.id)
+    cy.findByLabelText('Domain Secret').type(webhook.domainSecret)
+    cy.findByLabelText('Device ID').type(webhook.deviceId)
+
+    cy.findByRole('button', { name: 'Create akenza core webhook' }).click()
+
+    cy.location('pathname').should(
+      'eq',
+      `${Cypress.config('consoleRootPath')}/applications/${appId}/integrations/webhooks`,
+    )
```

Comment:
This test could render a blank page or an error and still pass.

Going from the realistic test paradigm, you would check whether the name of the new webhook is visible on this page. Also, you would check that there is no error present on the page. This maps to the user's expectations of the result page.